context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java

```diff
@@ -45,7 +41,8 @@ public Scope attach(Context toAttach) {
 
   @Override
   public Context current() {
-    return THREAD_LOCAL_STORAGE.get();
+    Context threadContext = THREAD_LOCAL_STORAGE.get();
+    return threadContext != null ? threadContext : Context.root();
```

Comment:
I think we can simply remove this line, we check for null in the Context and this is not necessarily.

Reply:
And add to the javadoc this behavior.

Reply:
Ok, if it's expected that `ContextStorage#current` can return `null` it feels like I should also add the `@Nullable` annotation so that developers can expect this.

Reply:
Are you suggesting to change this Javadoc on `ContextStorage#current`?  It implies that this method will never return `null`:

```
  /**
   * Returns the current {@link DefaultContext}. If no {@link DefaultContext} has been attached yet,
   * this will be the {@linkplain Context#root()} root context}.
   */
```

Also feels weird that it calls out `DefaultContext` given that's a package-private class.

Reply:
See here: https://github.com/HaloFour/opentelemetry-java/blob/1512fa1d754644b0c2de3f30a6039c848bfc1c03/context/src/main/java/io/opentelemetry/context/Context.java#L72

Reply:
That's why I'm asking for clarity, because if it's expected that `ContextStorage#current` can return `null` then the Javadocs on that method should be updated to reflect this, and the method should be marked as `@Nullable`.

Reply:
What is more

>  ThreadLocalContextStorage sets the ThreadLocal to Context.root() within the static constructor

this seems to be useless then and could be removed.

CC @anuraaga

Reply:
I've committed the suggested changes, which more or less reverses what the PR was intended to do so I've changed the description of the PR (if that's ok).